File: cpp_src/toggle_forward/src/toggle_client.cpp
#include <ros/ros.h>
#include <std_srvs/Empty.h>

int main(int argc, char ** argv) {
    ros::init(argc, argv, "toggle_client");
    ros::NodeHandle nh;

    ros::ServiceClient client = nh.serviceClient<std_srvs::Empty>("toggle_forward");

    ros::service::waitForService("toggle_forward");

    while (1) {
        std_srvs::Empty::Request req;
        std_srvs::Empty::Response resp;

        bool ok = client.call(req, resp);
        if (ok) {
            ROS_INFO("send toggle cmd");
        } else {
            ROS_ERROR("failed to send cmd");
        }

        sleep(2);

    }





    return 0;
}

File: pyhon_src/toggle_forward/scripts/toggle_client.py
#! /usr/bin/env python3

import rospy
from std_srvs.srv import Empty, EmptyRequest

def main():
    rospy.init_node("toggle_client")
    client = rospy.ServiceProxy("toggle_forward", Empty)

    rospy.wait_for_service("toggle_forward")

    rate = rospy.Rate(2)

    while True:
        try:
            req = EmptyRequest()
            resp = client(req)
            rospy.loginfo("send toggle cmd")



        except rospy.ServiceException as e:
            rospy.loginfo("send toggle cmd failed: %s" %e)

        rate.sleep()

if __name__ == "__main__":
    main()